src/main/java/com/google/devtools/build/lib/metrics/MetricsCollector.java

```diff
@@ -235,11 +253,34 @@ private BuildMetrics createBuildMetrics() {
     return buildMetrics.build();
   }
 
+  private ActionData buildActionData(ActionStats actionStats) {
+    NanosToMillisSinceEpochConverter nanosToMillisSinceEpochConverter =
+        BlazeClock.createNanosToMillisSinceEpochConverter();
+    ActionData.Builder builder = ActionData.newBuilder()
+        .setMnemonic(actionStats.mnemonic)
+        .setFirstStartedMs(
+            nanosToMillisSinceEpochConverter.toEpochMillis(
+                actionStats.firstStarted.longValue()))
+        .setLastEndedMs(
+            nanosToMillisSinceEpochConverter.toEpochMillis(
+                actionStats.lastEnded.longValue()))
+        .setActionsExecuted(actionStats.numActions.get());
+    if (actionStats.systemTime.sum().isPresent()) {
```

Comment:
Nit: I would prefer the result of `sum()` assigned to a variable, mainly to decrease verbosity, though also to make it clear that there can't be any race conditions.